python-avd/pyavd/_eos_designs/structured_config/network_services/application_traffic_recognition.py

```diff
@@ -45,192 +211,97 @@ def _wan_cp_app_dst_prefix(self: AvdStructuredConfigNetworkServicesProtocol) ->
-                    atr.applications.ipv4_applications.append(application_item)
-
-        # Applications in categories
-        for category in atr.categories:
-            for application in category.applications:
-                if application.name in self.inputs.application_classification.applications.ipv4_applications:
-                    application_item = self.inputs.application_classification.applications.ipv4_applications[application.name]
-                    atr.applications.ipv4_applications.append(application_item)
+        * the device Pathfinders vtep_ips as destination for non Pathfinders.
+        * the device Pathfinder vtep_ip as source.
 
-        self.structured_config.application_traffic_recognition.applications.ipv4_applications.extend(atr.applications.ipv4_applications)
+        Edge and Transit:
 
-        for application in atr.applications.ipv4_applications:
-            for prefix_set_name in (application.src_prefix_set_name, application.dest_prefix_set_name):
-                if not prefix_set_name:
-                    continue
+            application_traffic_recognition:
+              field_sets:
+                ipv4_prefixes:
+                  - name: PFX-PATHFINDERS
+                    prefix_values: [Pathfinder to which the router is connected vtep_ips]
 
-                if prefix_set_name not in self.inputs.application_classification.field_sets.ipv4_prefixes:
-                    msg = (
-                        f"The IPv4 prefix field set {prefix_set_name} used in the application {application} "
-                        "is undefined in 'application_classification.fields_sets.ipv4_prefixes'."
-                    )
-                    raise AristaAvdInvalidInputsError(msg)
+        Pathfinder:
 
-                ipv4_prefix_item = self.inputs.application_classification.field_sets.ipv4_prefixes[prefix_set_name]
-                atr.field_sets.ipv4_prefixes.append(ipv4_prefix_item)
+            application_traffic_recognition:
+              field_sets:
+                ipv4_prefixes:
+                  - name: PFX-LOCAL-VTEP-IP
+                    prefix_values: [Pathfinder vtep_ip]
 
-            for port_set_name in (
-                application.udp_src_port_set_name,
-                application.udp_dest_port_set_name,
-                application.tcp_src_port_set_name,
-                application.tcp_dest_port_set_name,
-            ):
-                if not port_set_name:
-                    continue
-                if port_set_name not in self.inputs.application_classification.field_sets.l4_ports:
-                    msg = (
-                        f"The L4 Ports field set {port_set_name} used in the application {application} "
-                        "is undefined in 'application_classification.fields_sets.l4_ports'."
-                    )
-                    raise AristaAvdInvalidInputsError(msg)
+        """
+        if self.shared_utils.is_wan_client:
```

Comment:
More readable and DRY if you set pathfinder_vtep_ips with either one or the other value and then return the model in one place.

Reply:
well thats not the same thing actually and the name is also different but refactored.